File: ilamicbook/backend-go/src/main/go/com/ilamicbook/backend/models/book.go
package models

import "time"

// Book represents a book in the library
type Book struct {
	ID          int       `json:"id"`
	Title       string    `json:"title"`
	Author      string    `json:"author"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at,omitempty"`
	UpdatedAt   time.Time `json:"updated_at,omitempty"`
}

// BookResponse represents the response structure for books
type BookResponse struct {
	Books []Book `json:"books"`
	Total int    `json:"total"`
}

File: ilamicbook/backend-node/src/main/javascript/com/ilamicbook/backend/models/book.js
/**
 * Book model representing a book in the library
 */
class Book {
    constructor(id, title, author, description, createdAt = new Date(), updatedAt = new Date()) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.description = description;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    /**
     * Convert to JSON object
     */
    toJSON() {
        return {
            id: this.id,
            title: this.title,
            author: this.author,
            description: this.description,
            created_at: this.createdAt,
            updated_at: this.updatedAt
        };
    }

    /**
     * Create Book from JSON object
     */
    static fromJSON(json) {
        return new Book(
            json.id,
            json.title,
            json.author,
            json.description,
            new Date(json.created_at),
            new Date(json.updated_at)
        );
    }
}

module.exports = Book; 